history-api/script.js

```diff
@@ -0,0 +1,43 @@
+// Update the page with the new content
+function displayContent(content) {
+  const description = document.querySelector("#description");
+  description.textContent = content.description;
+
+  const photo = document.querySelector("#photo");
+  photo.setAttribute("src", content.image.src);
+  photo.setAttribute("alt", content.image.alt);
+}
+
+// Handle click on link
+document.addEventListener("click", async (event) => {
+  const creature = event.target.getAttribute("data-creature");
+  if (creature) {
+    event.preventDefault();
+    try {
+      const response = await fetch(`creatures/${creature}.json`);
+      const json = await response.json();
+      displayContent(json);
+      history.pushState(json, "", creature);
```

Comment:
Adding a comment as this is the key point for the reader.
```suggestion
      // Add a new entry to the history.
      // This simulates loading a new page.
      history.pushState(json, "", creature);
```